k9mail/src/main/java/com/fsck/k9/helper/FileHelper.java

```diff
@@ -147,6 +141,11 @@ public static boolean move(final File from, final File to) {
 
         try {
             copyFile(from, to);
+
+            boolean deleteFromOk = from.delete();
```

Comment:
If you make it `deleteFailed` then you don't have to use the negation in the `if` below.